### ISSUE TITLE
Table: [A11y] Loss of focus and unable to access interactive content within each row in small screen size

### ISSUE THREAD

**Issue by sleric1024** [SAP/ui5-webcomponents#8385]
### Describe the bug

There is no way to tab inside each row to access the links and quick card within each row in small screen size for example 320px

### Isolated Example

https://stackblitz.com/edit/github-xbtmwe?file=src%2FApp.tsx

### Reproduction steps

1.Click the URL
2. Change the screen size to 320px
3. When using tab key for access the accessible elements from each rows, you will find it cannot be access inside for each row
...

### Expected Behaviour

All the accessible elements should be accessible from each rows

### Screenshots or Videos

<img width="744" alt="image" src="https://github.com/SAP/ui5-webcomponents-react/assets/5724592/9dcd3c78-f57e-44eb-8066-82806e0259c4">

### UI5 Web Components for React Version

~1.25.0

### UI5 Web Components Version

~1.22.0

### Browser

Chrome

### Additional Context

I am from SF Web team and this issue is reported by a11y exports while testing the projects that are using this component.

### Declaration

- [X] I’m not disclosing any internal or sensitive information.
